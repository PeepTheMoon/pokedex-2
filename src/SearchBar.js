import React, { Component } from 'react';
import request from 'superagent';
import PokemonItem from './PokemonItem.js';

export default class SearchAttackDefenseValue extends Component {

    state = {
        data: [],
        attack: 0,
        defense: 0,
        searchQuery: '',
        type: '',
        textSelection: 'pokemon', 
        statSelection: 'attack',
        page: 1
    }

    handleStatDropdown = (e) => {
        const value = e.target.value;
        this.setState({ statSelection: value })
    }

    handleTextDropdown = (e) => {
        const value = e.target.value;
        this.setState({ textSelection: value });
    }

    //searches by attack or defense value
    handleAttackDefenseChange = (e) => {
        const value = e.target.value;
        this.setState({ attack: value, defense: value });
    }

    //sends the value of the input to the api and waits for the returned data
    handleAttackDefenseClick = async () => {
        const fetchedAttackDefense = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?${this.state.statSelection}=${this.state.attack}`)â€©;
        //sets the state to the returned data and to the body of the page
        console.log(fetchedAttackDefense);
        this.setState({ data: fetchedAttackDefense.body.results })
    }

   handleChange= (e) => {
        const value = e.target.value;
        this.setState({ searchQuery: value, type: value });
    }

    handleClick = async () => {
        //sends the value of the input to the api and waits for the returned data
        const fetchedData = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?${this.state.textSelection}=${this.state.searchQuery}`);
        //sets the state to the returned data and to the body of the page
        console.log(fetchedData);
        this.setState({ data: fetchedData.body.results });
    } 

    render() {
        return (
            <div>
                <div className="dropdown-attack-defense">
                    <label>Choose Attack or Defense:  </label>
                    <select className="attack-defense" onChange={this.handleStatDropdown}>
                        <option value="attack">Attack</option>
                        <option value="defense">Defense</option>
                    </select>
                </div>
                
                 <div className="value-input">
                    <label>Search by Attack or Defense Value:  </label>
                    <input type="number" min="0" max="100" onChange={this.handleAttackDefenseChange} placeholder="Num"/>
                    <button onClick={this.handleAttackDefenseClick}>Search</button>
                </div>

                <div className="dropdown-name-type">
                    <label>Choose Name or Type:  </label>
                    <select className="name-type" onChange={this.handleTextDropdown}>
                        <option value="pokemon">Name</option>
                        <option value="type">Type</option>
                    </select>
                </div>

                <div className="word-search">
                    <label>Search by Name or Type:  </label>
                    <input className="pokemon-search" onChange={this.handleChange} placeholder="Enter Name or Type"/>
                    <button onClick={this.handleClick}>Search</button>
                </div>

                <ul>
                    {
                        this.state.data.map((character) => {
                            return <div className="pokemon-card">
                                    <PokemonItem character={character}/>
                                    </div>
                        })
                    } 
                </ul>
            </div>
        )
    }
}
