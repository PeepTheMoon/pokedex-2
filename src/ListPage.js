import React, { Component } from 'react';
import request from 'superagent';
import SearchBar from './SearchBar.js';

export default class ListPage extends Component {

//initializes state
    state = {
        data: [],
        searchQuery: '',
        type: '',
        attack: 0,
        defense: 0,
        statSelection: 'attack',
        textSelection: 'pokemon',
        page: 1
    }

    async componentDidMount(){
        const searchParams = new URLSearchParams(window.location.search);
        const query = searchParams.get('search') || '';
        this.setState({ searchQuery: query });
        
        if (query){
          let page = 1;

          if (searchParams.get('page')){
            page = searchParams.get('page');
          }

        const response = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?pokemon=${query}&page=${page}`)
        const results = response.body.results;
        this.setState({ data: results })  

      } else {

          const response = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex`);
          const results = response.body.results;
          this.setState({ data: results })
        }
      }

          //allows user to view the next page of results
    toNextPage = async () => {

        const nextPage = this.state.page + 1; 
        this.setState({ page: nextPage });

        const response = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?${this.state.statSelection}=${this.state.attack}&page=${nextPage}`) ;
        const results = response.body.results;
        this.setState({ data: results });
    }

    //allows user to view the previous page of results
    toPreviousPage = async () => {
        const previousPage = this.state.page - 1;  
        this.setState({ page: previousPage }) 
      
        const response = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?${this.state.statSelection}=${this.state.attack}&page=${previousPage}`) ;
        const results = response.body.results;
        this.setState({ data: results })
      }

    render() {
        return (
            <div>
                <section className="searchbar">
                    <SearchBar />
                </section>
                 <div className="paging-buttons">
                {this.state.page > 1 && <button onClick={this.toPreviousPage}>Previous</button>}
              
                <button onClick={this.toNextPage}>Next</button>
              </div>
            </div>
        )
    }
}


