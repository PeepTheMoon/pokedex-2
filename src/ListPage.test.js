import React from 'react';
import { shallow } from 'enzyme';
import ListPage from './ListPage.js';
import SearchBar from './SearchBar.js';

it('renders single pokemon item details', () => {
    const wrapper = shallow(<SearchBar/>, <ListPage character={{ pokemon: 'character',url_image: 'some image', type_1: 'some type', attack: 0, defense: 0}}/>);
  
    expect(wrapper).toMatchSnapshot();
  });