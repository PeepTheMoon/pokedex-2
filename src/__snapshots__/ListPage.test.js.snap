// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders single pokemon item details 1`] = `
<div>
  <div
    className="dropdown-attack-defense"
  >
    <label>
      Choose Attack or Defense:  
    </label>
    <select
      className="attack-defense"
      onChange={[Function]}
    >
      <option
        value="attack"
      >
        Attack
      </option>
      <option
        value="defense"
      >
        Defense
      </option>
    </select>
  </div>
  <div
    className="value-input"
  >
    <label>
      Search by Attack or Defense Value:  
    </label>
    <input
      max="100"
      min="0"
      onChange={[Function]}
      placeholder="Num"
      type="number"
    />
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <div
    className="dropdown-name-type"
  >
    <label>
      Choose Name or Type:  
    </label>
    <select
      className="name-type"
      onChange={[Function]}
    >
      <option
        value="pokemon"
      >
        Name
      </option>
      <option
        value="type"
      >
        Type
      </option>
    </select>
  </div>
  <div
    className="word-search"
  >
    <label>
      Search by Name or Type:  
    </label>
    <input
      className="pokemon-search"
      onChange={[Function]}
      placeholder="Enter Name or Type"
    />
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <ul />
</div>
`;
